### PR TITLE
Keyboard [null] ST110R2.1 (SaikouType) Full Size ISO support

### PR DESCRIPTION
**Pull request by user176176**
## Description

Basic support for [null] ST110R2.1 keyboard with ISO layout

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/null/st110r2/info.json [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        "backlight": false,
+        "unicode": false
+    },
+    "matrix_size": {"rows": 8, "cols": 15},
+    "matrix_pins": {
+        "cols": ["F0", "F1", "F2", "F3", "F4", "F5", "F6", "F7", "C0", "C1", "C2", "C3", "C4", "C5", "C6"],
+        "rows": ["A0", "A1", "A2", "A3", "A4", "A5", "A6", "A7"]
+    },
+    "processor": "at90usb1286",
+    "url": "",
```
Comment (dexter93) [inline keyboards/null/st110r2/info.json:27]:
```suggestion
```
duplicate attribute, remove

File: keyboards/null/st110r2/info.json [hunk 2]
```diff
@@ -0,0 +1,155 @@
+{
+    "manufacturer": "qmk",
```
Comment (dexter93) [inline keyboards/null/st110r2/info.json:2]: this should probably be something other than "qmk". looking at the url provided it's probably 
```suggestion
    "manufacturer": "[null]",
```

File: keyboards/null/st110r2/config.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Copyright 2023 qmk (@qmk)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
```
Comment (dexter93) [inline keyboards/null/st110r2/config.h:4]: if the whole `config.h` is blank, just omit it

File: keyboards/null/st110r2/info.json [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    "keyboard_name": "null/st110r2",
```
Comment (dexter93) [inline keyboards/null/st110r2/info.json:3]:
```suggestion
    "keyboard_name": "ST110 r2",
```

File: keyboards/null/st110r2/info.json [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "audio": false,
+        "midi": false,
+        "rgblight": false,
+        "backlight": false,
+        "unicode": false
+    },
+    "matrix_size": {"rows": 8, "cols": 15},
```
Comment (dexter93) [inline keyboards/null/st110r2/info.json:21]: `matrix_size` should be automatically calculated by the layout.. why do you need to specify it here?

File: keyboards/null/st110r2/readme.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+![st110r2_zoom](https://s3.amazonaws.com/null-src/images/posts/nkbm-st110r2/NKBM-ST110r21-both1.png)
+![st110r2_full](https://s3.amazonaws.com/null-src/images/posts/nkbm-st110r2/NKBM-ST110r21-both3.png)
+
+* Keyboard Maintainer: [user176176](https://github.com/user176176)
+* Hardware Supported: [null] 最高タイプ (SaikouType) NKBM-ST110r2.x keyboard PCBs with Teensy 2.0++.
+* Hardware Availability: [NKBM-ST110r2 - null-src.com](https://null-src.com/posts/nkbm-st110r2/post.php)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make null/st110r2:default
```
Comment (dexter93) [inline keyboards/null/st110r2/readme.md:14]:
```suggestion
    qmk compile -kb null/st110r2 -km default
```
is the new way

File: keyboards/null/st110r2/readme.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+Flashing example for this keyboard:
+
+    make null/st110r2:default:flash
```
Comment (dexter93) [inline keyboards/null/st110r2/readme.md:18]: see above comment, also `qmk flash`

**Review by dexter93 — CHANGES_REQUESTED** [review]

File: keyboards/null/st110r2/info.json [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0000",
+        "vid": "0xFEED"
+    },
+    "indicators": {
+        "num_lock": "B7",
+        "caps_lock": "D0",
+        "scroll_lock": "D1"
+    },
```
Comment (dunk2k) [inline keyboards/null/st110r2/info.json:37]:
```suggestion
    },
    "community_layouts": ["fullsize_iso"],
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/null/st110r2/info.json (hunk 5 above)
Comment (user176176) [reply keyboards/null/st110r2/info.json:21]: Good catch. I must have looked it up from that not data driven qmk fork on the product page that I used as the learning material for this.

File: keyboards/null/st110r2/readme.md (hunk 6 above)
Comment (zvecr) [reply keyboards/null/st110r2/readme.md:14]: This recommendation is currently not required.

Comment (user176176) [reply keyboards/null/st110r2/readme.md:14]: I actually used the CLI / qmk compile and qmk flash but the qmk new-keyboard had these as defaults so I didn't dare to change the "official way of doing this".

**Pull request closed without merging**